File: playground-master/go_php/3-refactor/refactor.php
<?php

function findFirstStringInBracket($str)
{
    if (strlen($str) > 0) {
        $pattern = '#\((.*?)\)#';
        echo 'Pattern: ', $pattern, "\n";
        
        preg_match($pattern, $str, $match);
        return $match[1];

    } else {
        return '';
    }
}

$line_test = "(hehe) (haha)";
$result = findFirstStringInBracket($line_test);
assert($result, "hehe");
echo $result, "\n";

$line_test = "hehe) (haha)";
$result = findFirstStringInBracket($line_test);
assert($result, "haha");
echo $result, "\n";

?>

File: playground-master/go_php/3-refactor/refactor.go
package main

import (
	"fmt"
	"regexp"
	"strings"
)

func findFirstStringInBracket(str string) string {
	var element string
	if (len(str)) > 0 {
		re := regexp.MustCompile(`\((.*?)\)`)
		fmt.Printf("Pattern: %v\n", re.String()) // print pattern

		submatchall := re.FindAllString(str, -1)
		element = strings.Trim(submatchall[0], "(") //submatchall[0] is first element of array
		element = strings.Trim(element, ")")
	} else {
		return ""
	}

	return element
}

func main() {
	line_test := "(hehe) (haha)"
	result := findFirstStringInBracket(line_test)
	fmt.Println(result)

	line_test = "hehe) (haha)"
	result = findFirstStringInBracket(line_test)
	fmt.Println(result)

}